client/ayon_core/lib/path_tools.py

```diff
@@ -136,8 +136,8 @@ def version_up(filepath):
         index += len(new_label)
         clash_basename = clash_basename[:index]
 
-    for file in os.listdir(dirname):
-        if file.endswith(ext) and file.startswith(clash_basename):
+    for f in os.scandir(dirname):
+        if f.name.endswith(ext) and f.name.startswith(clash_basename):
```

Comment:
I personally think it raising an error if the folder does not exist is fine behavior.